add alert-link styles to anchor

5h1rU opened the pull request:
![Screen Shot 2021-04-19 at 3 11 58 PM](https://user-images.githubusercontent.com/989826/115297276-a4742580-a121-11eb-82a5-dd0fe22d6306.png)
![Screen Shot 2021-04-19 at 3 12 10 PM](https://user-images.githubusercontent.com/989826/115297280-a5a55280-a121-11eb-8606-288dfb36cef1.png)


<!-- Reminder: Have you updated the changelog and relevant docs (user docs, architecture diagram, etc) ? -->

Fixes #20152

5h1rU commented on client/branded/src/global-styles/alert.scss:
```diff
@@ -1,4 +1,5 @@
 @import 'bootstrap/scss/alert';
+@import 'bootstrap';
```
not sure if this is the best way to get the `alert-link` colors

felixfbecker replied:
The `.alert-link` class should be imported by the import line right above this, no?
`@import 'bootstrap'` brings in the whole library, we generally only import selectively what we use, so we shouldn't do that.

limitedmage replied:
This change is causing significant visual regressions in spacing/text size throughout the whole webapp.

felixfbecker replied:
It probably imports the CSS dist bundle, instead of Sass, which means our variables and config is not respected (but I think we don't need this import at all)

5h1rU replied:
@felixfbecker bootstrap is generating those colors with a [SASS mixing](https://sourcegraph.com/github.com/twbs/bootstrap@main/-/blob/scss/mixins/_alert.scss#L1:20)

I need to figure out how to use the mixing since import the whole lib is a bad idea.

5h1rU replied:
Ok, validating again as @felixfbecker said the import isn't necessaryy

felixfbecker commented on client/branded/src/global-styles/alert.scss:
```diff
@@ -19,3 +20,11 @@
     text-decoration: underline;
 }
+
+.alert.alert-info,
+.alert.alert-warning,
+.alert.alert-danger {
```
Is there a reason we can't just match on `.alert`? E.g. do you specifically want to exclude `.alert-secondary`? I think it should apply to all alerts
```suggestion
.alert {
```

felixfbecker replied:
Since this doesn't use `:where()`, did you test the specificity problem I described in https://github.com/sourcegraph/sourcegraph/issues/20152#issuecomment-822687474? I'm not sure if it's actually a problem, so if you tested that and it was no issue seems fine, but if not I am relatively sure there would be an issue here. As mentioned I think the easiest thing to do would be to add an alert with a button link to the story.

felixfbecker commented:
As part of this change it would be good to update the global story for alerts to remove the `alert-link` classes - that way we can verify that it works correctly (and prevent regressions). If I go into that story and remove the class manually in dev tools, I don't get the desired style yet.

5h1rU commented:
> As part of this change, it would be good to update the global story for alerts to remove the `alert-link` classes - that way, we can verify that it works correctly (and prevent regressions). If I go into that story and remove the class manually in dev tools, I don't get the desired style yet.

@felixfbecker, I added all the variants on the storybook and removed the class name. Now we have an underline decoration for all the link alerts that weren't there before. @rrhyne can you confirm if the underlined link is the expected behavior?

felixfbecker commented:
It was easier for me to make the suggestions I wanted to make through a PR than through comments, please see here: https://github.com/sourcegraph/sourcegraph/pull/20163

rrhyne commented:
Looks good to me, thanks for getting on this so quickly!

eseliger reviewed (approved):
Should we remove remaining instances of `.alert-link` in the codebase, too?

eseliger commented:
Actually, I think the behavior here is changed: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/web/src/extensions/ExtensionCard.tsx#L188:1-191:34

5h1rU commented:
@eseliger, Can you please review the Percy visual changes? Batch-changes are showing some differences. I want to know if those changes are expected or need to be addressed. CC @rrhyne

eseliger commented:
Hmm I don't have a strong opinion, if this is the new standard for links in a card, this lgtm. Better like that than having a custom override to retain the previous style there 🤷‍♂️

rrhyne commented:
I looked through the batch changes and approved them. This standardizes the current solution. The design refresh will alter these links to be a more recognizable blue as batch changes previously used, so we'll come full circle.
